### PR TITLE
FB BatchSource Plugin

### PR DESCRIPTION
**Pull request by amolkokane-cirus**
FB BatchSource Plugin

### CODE REVIEW

**Comment by amolkokane-cirus** [comment]
@googlebot I signed it!

**Comment by shiva-cirus** [comment]
@googlebot I consent.

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+      <id>sonatype-snapshots</id>
+      <url>https://oss.sonatype.org/content/repositories/snapshots</url>
+    </repository>
+  </repositories>
+
+  <dependencies>
+    <dependency>
+      <groupId>io.cdap.cdap</groupId>
+      <artifactId>cdap-etl-api</artifactId>
+      <version>${cdap.version}</version>
```
Comment (rmstar) [inline pom.xml:59]: scope should be provided
https://github.com/data-integrations/google-cloud/blob/develop/pom.xml#L148
Please take a look at above pom file for scope of the artifacts

File: src/test/java/io/cdap/plugin/facebook/etl/FacebookBatchSourceTest.java [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+      .setBidAmount(100L)
+      .setStartTime(String.format("%s-01-01T00:00:00+0000", year))
+      .setEndTime(String.format("%s-01-01T00:00:00+0000", year + 1))
+      .setOptimizationGoal("REACH")
+      .setTargeting("{\"geo_locations\":{\"countries\":[\"US\"]},\"publisher_platforms\":[\"facebook\"]}")
+      .execute();
+
+    adSets.add(result);
+
+    return result;
```
Comment (rmstar) [inline src/test/java/io/cdap/plugin/facebook/etl/FacebookBatchSourceTest.java]: return value is unused?

**Review by rmstar — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (amolkokane-cirus) [reply pom.xml:59]: Done

File: src/test/java/io/cdap/plugin/facebook/etl/FacebookBatchSourceTest.java (hunk 2 above)
Comment (amolkokane-cirus) [reply src/test/java/io/cdap/plugin/facebook/etl/FacebookBatchSourceTest.java]: Done

File: src/main/java/io/cdap/plugin/facebook/source/common/AdsInsightsTransformer.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    JsonObject insightsJson = insights.getRawResponseAsJsonObject();
+    insightsJson.entrySet().forEach(entry -> {
+      if (schemaContainsField(schema, entry.getKey())) {
+        Schema fieldSchema = schema.getField(entry.getKey()).getSchema();
+
+        if (fieldSchema.isNullable()) {
+          fieldSchema = fieldSchema.getNonNullable();
+        }
+
+        switch (fieldSchema.getType()) {
```
Comment (CuriousVini) [inline src/main/java/io/cdap/plugin/facebook/source/common/AdsInsightsTransformer.java:50]: why other types are not supported like integer, double, date, timestamp etc

**Review by CuriousVini — APPROVED** [review]

File: src/main/java/io/cdap/plugin/facebook/source/common/AdsInsightsTransformer.java (hunk 3 above)
Comment (shiva-cirus) [reply src/main/java/io/cdap/plugin/facebook/source/common/AdsInsightsTransformer.java:50]: We are not sure why the original developers not support those types. Let me know if you want me to add a TO DO here. Facebook Ads plugins needs a lot of changes and is not very useful in the correct form. This feedback was provided to Bhooshan from our testing also also from P&G testing.

Comment (CuriousVini) [reply src/main/java/io/cdap/plugin/facebook/source/common/AdsInsightsTransformer.java:50]: Please add TODO with corresponding jira

Comment (shiva-cirus) [reply src/main/java/io/cdap/plugin/facebook/source/common/AdsInsightsTransformer.java:50]: Code Updated with TODO and created JIRA ticket - https://issues.cask.co/browse/PLUGIN-365

File: src/main/java/io/cdap/plugin/facebook/source/batch/FacebookInputFormat.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * InputFormat for mapreduce job, which provides a single split of data.
+ */
+public class FacebookInputFormat extends InputFormat {
+  @Override
+  public List<InputSplit> getSplits(JobContext jobContext) {
+    return Collections.singletonList(new FacebookSplit());
```
Comment (CuriousVini) [inline src/main/java/io/cdap/plugin/facebook/source/batch/FacebookInputFormat.java:39]: This is using single split. Could you create a jira for it and add a TODO with the jira number here?

**Pull request merged**